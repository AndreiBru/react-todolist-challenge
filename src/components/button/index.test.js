import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from './';

const types = ['add', 'edit', 'default'];

describe('Button', () => {
  types.forEach(type => {
    it(`renders and matches snapshot for type=${type}`, () => {
      const comp = renderer.create(
        <Button type={type}>Button type {type}</Button>,
      );

      expect(comp.toJSON()).toMatchSnapshot();
    });
  });

  it('calls the onClick handler', () => {
    const onClickSpy = jest.fn();
    const comp = shallow(
      <Button type="edit" onClick={onClickSpy}>
        Click me
      </Button>,
    );

    expect(onClickSpy).not.toHaveBeenCalled();
    comp.find('button').simulate('click');
    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
});
