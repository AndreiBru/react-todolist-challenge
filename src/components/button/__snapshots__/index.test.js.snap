// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders and matches snapshot for type=add 1`] = `
<button
  className="button add"
  onClick={undefined}
>
  Button type 
  add
</button>
`;

exports[`Button renders and matches snapshot for type=default 1`] = `
<button
  className="button default"
  onClick={undefined}
>
  Button type 
  default
</button>
`;

exports[`Button renders and matches snapshot for type=edit 1`] = `
<button
  className="button edit"
  onClick={undefined}
>
  Button type 
  edit
</button>
`;
